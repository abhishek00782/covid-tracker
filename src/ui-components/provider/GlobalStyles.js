import { createGlobalStyle } from 'styled-components';

const GlobalStyles = createGlobalStyle`
    *, *:before, *:after {
        box-sizing: border-box;
    }
    * {
        margin: 0;
        padding: 0;
    }
    .disable-scroll {
        overflow: hidden;
    }
`;

export default GlobalStyles;
