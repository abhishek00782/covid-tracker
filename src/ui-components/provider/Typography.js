import { createGlobalStyle } from "styled-components";

const Typography = createGlobalStyle`
    body, input, textarea, button {
        -webkit-font-smoothing: antialiased;
        -moz-osx-font-smoothing: grayscale;
        font-family: "Inter", sans-serif;
        font-size: 13px;
        font-weight: 400;
        line-height: 20px;
        letter-spacing: 0;
        color: ${(props) => props.theme.colors.TEXT};
    }
    a {
        text-decoration: none;
        color: ${(props) => props.theme.colors.TEXT_ACTION};
    }
    strong, button, b {
        font-weight: 700;
    }
    small {
        font-size: 11px;
        letter-spacing: 0.15px;
        line-height: 16px;
    }
    label, .light {
        color: ${(props) => props.theme.colors.TEXT_LIGHT_1};
    }
    h1, h2, h3 {
        font-weight: 500;
        letter-spacing: -0.15px;
    }
    h3 {
        font-size: 15px;
        line-height: 24px;
    }
    h2 {
        font-size: 17px;
        line-height: 28px;
    }
    h1 {
        font-size: 19px;
        line-height: 28px;
    }
`;

export default Typography;
