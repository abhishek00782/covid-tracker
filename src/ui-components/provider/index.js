import React from "react";
import { ThemeProvider } from "styled-components";
import colors from "../colors";
import Typography from "./Typography";
import GlobalStyles from "./GlobalStyles";

const defaultTheme = {
  colors,
};

export const Provider = ({ theme = defaultTheme, children }) => (
  <ThemeProvider theme={theme}>
    <Typography />
    <GlobalStyles />
    {children}
  </ThemeProvider>
);
