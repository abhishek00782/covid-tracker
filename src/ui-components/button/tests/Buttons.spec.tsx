import React from "react";
import renderer from "react-test-renderer";

import { Button } from "..";
import { Provider } from "./../../provider";

describe("<Button />", () => {
  const wrapper = renderer.create(
    <Provider>
      <Button>Click Me</Button>
    </Provider>
  );

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
