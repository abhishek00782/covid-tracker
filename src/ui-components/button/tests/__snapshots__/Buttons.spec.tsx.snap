// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render 1`] = `
Array [
  <div
    className="hfMfKm"
  >
    <div />
  </div>,
  <button
    className="gtEVgi"
  >
    Click Me
  </button>,
]
`;
